internal/provider/resource_local_sensitive_file_test.go

```diff
@@ -0,0 +1,216 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	r "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestLocalSensitiveFile_Basic(t *testing.T) {
+	td := createTestingTempDir(t)
+	defer os.RemoveAll(td)
+
+	f := filepath.Join(td, "local_file")
+	f = strings.ReplaceAll(f, `\`, `\\`)
+
+	var cases = []struct {
+		path    string
+		content string
+		config  string
+	}{
+		{
+			f,
+			"This is some content", fmt.Sprintf(`
+				resource "local_file" "file" {
+				  content  = "This is some content"
+				  filename = "%s"
+				}`, f,
+			),
+		},
+		{
+			f,
+			"This is some sensitive content", fmt.Sprintf(`
+				resource "local_file" "file" {
+				  sensitive_content = "This is some sensitive content"
+				  filename = "%s"
+				}`, f,
+			),
+		},
+		{
+			f,
+			"This is some base64 content", fmt.Sprintf(`
+				resource "local_file" "file" {
+				  content_base64 = "VGhpcyBpcyBzb21lIGJhc2U2NCBjb250ZW50"
+				  filename = "%s"
+				}`, f,
+			),
+		},
+		{
+			f,
+			"This is some base64 content", fmt.Sprintf(`
+				resource "local_file" "file" {
+				  content_base64 = base64encode("This is some base64 content")
+				  filename = "%s"
+				}`, f,
+			),
+		},
+	}
+
+	for i, tt := range cases {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			r.UnitTest(t, r.TestCase{
+				Providers: testProviders,
+				Steps: []r.TestStep{
+					{
+						Config: tt.config,
+						Check: func(s *terraform.State) error {
+							content, err := ioutil.ReadFile(tt.path)
+							if err != nil {
+								return fmt.Errorf("config:\n%s\n,got: %s\n", tt.config, err)
+							}
+							if string(content) != tt.content {
+								return fmt.Errorf("config:\n%s\ngot:\n%s\nwant:\n%s\n", tt.config, content, tt.content)
+							}
+							return nil
+						},
+					},
+				},
+				CheckDestroy: func(*terraform.State) error {
+					if _, err := os.Stat(tt.path); os.IsNotExist(err) {
+						return nil
+					}
+					return errors.New("local_file did not get destroyed")
+				},
+			})
+		})
+	}
+}
+
+func TestLocalSensitiveFile_source(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "local-file-source")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
+
+	// create a local file that will be used as the "source" file
+	source_content := "local file content"
+	if err := ioutil.WriteFile("source_file", []byte(source_content), 0644); err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove("source_file")
+
+	config := `
+		resource "local_file" "file" {
```

Comment:
This one too.

Reply:
Oh gee, I missed to rename a bunch of resources. I blame cut & pasting.

Thank you @jhancock93 :)